common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleServiceMetrics.java

```diff
@@ -78,7 +82,32 @@ public static void collectMetric(
           new ShuffleServiceMetricsInfo(name + "_rate1", "1 minute rate of timer " + name),
           t.getOneMinuteRate())
         .addGauge(new ShuffleServiceMetricsInfo(name + "_rateMean", "Mean rate of timer " + name),
-          t.getMeanRate());
+          t.getMeanRate())
+        .addGauge(
+          getShuffleServiceMetricsInfoForGenericValue(timingName, "max"), snapshot.getMax())
+        .addGauge(
+          getShuffleServiceMetricsInfoForGenericValue(timingName, "min"), snapshot.getMin())
+        .addGauge(
+          getShuffleServiceMetricsInfoForGenericValue(timingName, "mean"), snapshot.getMean())
+        .addGauge(
+          getShuffleServiceMetricsInfoForGenericValue(timingName, "stdDev"), snapshot.getStdDev());
+      for (int percentileThousands : new int[] { 10, 50, 250, 500, 750, 950, 980, 990, 999 }) {
+        String percentileStr;
+        switch (percentileThousands) {
+          case 10:
+            percentileStr = "1stPercentile";
+            break;
+          case 999:
+            percentileStr = "99.9thPercentile";
```

Comment:
Hi, @xkrogen . Thank you for adding this and I saw your [comment](https://github.com/apache/spark/pull/32388#discussion_r629507805).

Technically, `999thPercentile` is consistent with the conventional in Apache Spark instead of `99.9thPercentitle`. If you look at the other existing metrics, it has the following.
<img width="146" alt="Screen Shot 2021-05-30 at 6 52 14 PM" src="https://user-images.githubusercontent.com/9700541/120129072-325b1b80-c178-11eb-9ff2-dd242951e24c.png">


And, just a question. May I ask where you borrow this `99.9thPercentitle`?

Reply:
Thanks for sharing, I did not realize there was an existing convention here. I've updated it to be `999thPercentile`. I didn't borrow it from anywhere in particular, just how I would say it aloud: "ninety-nine point ninth percentile".